libecole/src/observation/nodebipartite.cpp

```diff
@@ -355,7 +377,14 @@ utility::coo_matrix<value_type> extract_edge_features(scip::Model& model) {
 
 auto NodeBipartite::extract(scip::Model& model, bool /* done */) -> std::optional<NodeBipartiteObs> {
 	if (model.get_stage() == SCIP_STAGE_SOLVING) {
-		return NodeBipartiteObs{extract_col_features(model), extract_row_features(model), extract_edge_features(model)};
+		if (use_cache) {
+			if (is_on_root_node(model)) {
```

Comment:
What happens if someone uses this observation function with an environment that skips the root node ? I guess it's ok to tell people they shouldn't , but in that case this will trigger a SEGFAULT from what I understand from the code. Could we handle that more gently, maybe with an Exception ?

Reply:
I had more in mind that this would be a partial (experimental). The full solution should handle the cache better as we discussed.

What would you add here? A boolean flag that the root node as been visited, and check it when before extracting?

Reply:
Yes, a simple check like that. But as you said, other checks are necessary,
like making sure no cuts have been added since the cache creation. But I
guess we can merge as-is and just open an issue regarding the things that
remain to be done for safety.